x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/googlevertexai/embeddings/GoogleVertexAiEmbeddingsTaskSettings.java

```diff
@@ -84,11 +142,16 @@ public TransportVersion getMinimalSupportedVersion() {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalBoolean(this.autoTruncate);
+        out.writeOptionalEnum(this.inputType);
```

Comment:
This `writeTo()` method and the corresponding constructor `GoogleVertexAiEmbeddingsTaskSettings(StreamInput in)` are used for inter-node communication. This method writes the class settings out in a serialisable format which is then sent to anther node in cluster where it is read with the `StreamInput` constructor. 

The complication is that in a cluster of mixed versions the receiving node may be older and not know about the new  `inputType` field. To prevent errors the code needs to check if the receiving node is knows about the new field and we do that by checking the `TransportVersion`.

First add a new TransportVersion `VERTEX_AI_INPUT_TYPE_ADDED` to the end of the list in [TransportVersions](https://github.com/elastic/elasticsearch/blob/fa6c5296d40136c4ea7b559b34dc0e3fe775815f/server/src/main/java/org/elasticsearch/TransportVersions.java#L190).

Then, in this code add a check on the version


```suggestion
        if (out.getTransportVersion().onOrAfter(TransportVersions.VERTEX_AI_INPUT_TYPE_ADDED)) {
            out.writeOptionalEnum(this.inputType);
        }
```

And make the same check in the StreamInput constructor
```
        if (in.getTransportVersion().onOrAfter(TransportVersions.VERTEX_AI_INPUT_TYPE_ADDED)) {
            this.inputType = in.readOptionalEnum();
        } else {
            this.inputType = null;
        }
```

You will see this pattern all over the Elasticsearch codebase

Reply:
The serialisation logic is tested in `GoogleVertexAiEmbeddingsTaskSettingsTests` which extends `AbstractBWCWireSerializationTestCase`. `AbstractBWCWireSerializationTestCase` tests the serialisation logic all you need to do is update the `createRandom()` method to add the new field  - which you have done already

Reply:
Thank you for explaining! Updated in https://github.com/elastic/elasticsearch/pull/116431/commits/fcf7fff078f89fdd5cbd436416f40123a3657c08